src/Illuminate/Database/Query/Grammars/MySqlGrammar.php

```diff
@@ -50,6 +50,31 @@ protected function whereNotNull(Builder $query, $where)
         return parent::whereNotNull($query, $where);
     }
 
+    /**
+     * Compile a "where fulltext" clause.
+     *
+     * @param  \Illuminate\Database\Query\Builder  $query
+     * @param  array  $where
+     * @return string
+     */
+    public function whereFulltext(Builder $query, $where)
+    {
+        $columns = $this->columnize($where['columns']);
+
+        $query->addBinding($where['value']);
```

Comment:
We never add bindings in grammars typically. How will this binding be in the right order if where clauses are added after this operation while the developer is building the query?

Reply:
Hmm, i don't know how to solve it. The PostgreSQL version needs more user parameters than the MySQL one because a language name is required. I built the code with the idea in mind that the language parameter should be sql-injection safe. If we remove this security-barrier then the bug could be fixed. What do you think about it @taylorotwell @driesvints?

By the way it's a good example for my pull request #40115. With the grammar now having access to a sql-injection safe `quoteString` implementation, the added bindings in the grammar would not be needed anymore. The language parameter could be made safe by `quoteString` and the value binding moved to the `whereFulltext` method again.

So i see two solutions on fixing it:
1. Make the language parameter sql-injection vulnerable and move the binding of the search value to `whereFulltext`
2. Merging #40115 and moving fulltext search to `9.x` as the grammar could now use the safe `quoteString` method
3. Any other idea?

Reply:
I think it should be save to add the language param directly to the query since you're explicitly passing it as an option and not as a userland provided value. Sanitization should be done on the userside. @taylorotwell do you agree with that? If so, @tpetry could you send in those changes or do you want me to do that?

Reply:
I can work on it in a few hours.

Reply:
@tpetry I've already pushed the changes, thanks 👍